hyperdome_client/threads.py

```diff
@@ -72,20 +72,232 @@ def run(self):
+        super(EndChatTask, self).__init__()
+        self.session = session
+        self.server = server
+        self.uid = uid
+
+    def run(self):
+        try:
+            self.session.post(f"{self.server.url}/counseling_complete",
+                              data={'user_id': self.uid})
+            self.signals.success.emit('good')
+        except:
+            self.signals.error.emit("you're stuck here now")
+
+
+class CounselorSignoutTask(QtCore.QRunnable):
+    """
+    deregister counselor identified from active list,
+    stopping them from receiving additional guests
+    """
+    signals = TaskSignals()
+
+    def __init__(self, session: requests.Session, server: Server, uid: str):
+        super(CounselorSignoutTask, self).__init__()
+        self.session = session
+        self.server = server
+        self.uid = uid
+
+    def run(self):
+        try:
+            self.session.post(f"{self.server.url}/counselor_signout",
+                              data={"user_id": self.uid})
+            self.signals.success.emit('good')
+        except:
+            self.signals.error.emit("you're stuck here now")
+
+
+def send_message(server: Server,
+                 session: requests.Session,
+                 uid: str,
+                 message: str):
+    """
+    Send message to server provided using session for given user
+    """
+    session.post(
+        f'{server.url}/send_message',
+        data={
+            'message': message,
+            'user_id': uid})
+
+
+def get_uid(server: Server,
+            session: requests.Session):
+    """
+    Ask server for a new UID for a new user session
+    """
+    if server.is_counselor:
+        uid = session.post(f"{server.url}/counselor_signin",
+                           data={"username": server.username,
+                                 "password": server.password}).text
+    else:
+        uid = session.get(
+            f'{server.url}/generate_guest_id').text
+    return uid
+
+
+def get_messages(server: Server,
+                 session: requests.Session,
+                 uid: str):
+    """
+    collect new messages waiting on server for active session
+    """
+    return session.get(
+        f"{server.url}/collect_messages",
+        data={"user_id": uid}).text
+
+
+def start_chat(server: Server,
+               session: requests.Session,
+               uid: str):
+    if server.is_counselor:
+        return session.get(f"{server.url}/counselor_signin",
+                           data={"username": server.username,
+                                 "password": server.password}).text
+
+    else:
+        return session.post(
+            f"{server.url}/request_counselor",
+            data={"guest_id": uid}).text
+
+
+COMPATIBLE_SERVERS = ['2.0']
+
+
+def probe_server(server: Server,
+                 session: requests.Session):
+    info = json.loads(session.get(f"{server.url}/probe").text)
+    if not info['name'] == 'hyperdome':
```

Comment:
`if not info['name'] == 'hyperdome':` -> `if info['name'] != 'hyperdome':`

Reply:
Better.

Fixed.